src/coreclr/jit/importer.cpp

```diff
@@ -4782,6 +4788,415 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
+        }
+
+        case NI_SRCS_UNSAFE_Copy:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ldarg.1
+            // ldobj !!T
+            // stobj !!T
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_CopyBlock:
+        {
+            assert(sig->sigInst.methInstCount == 0);
+
+            // ldarg.0
+            // ldarg.1
+            // ldarg.2
+            // cpblk
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_CopyBlockUnaligned:
+        {
+            assert(sig->sigInst.methInstCount == 0);
+
+            // ldarg.0
+            // ldarg.1
+            // ldarg.2
+            // unaligned. 0x1
+            // cpblk
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_InitBlock:
+        {
+            assert(sig->sigInst.methInstCount == 0);
+
+            // ldarg.0
+            // ldarg.1
+            // ldarg.2
+            // initblk
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_InitBlockUnaligned:
+        {
+            assert(sig->sigInst.methInstCount == 0);
+
+            // ldarg.0
+            // ldarg.1
+            // ldarg.2
+            // unaligned. 0x1
+            // initblk
+            // ret
+
+            return nullptr;
+        }
+
+        case NI_SRCS_UNSAFE_IsAddressGreaterThan:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ldarg.1
+            // cgt.un
+            // ret
+
+            GenTree* op2 = impPopStack().val;
+            GenTree* op1 = impPopStack().val;
+
+            GenTree* tmp = gtNewOperNode(GT_GT, TYP_INT, op1, op2);
+            tmp->gtFlags |= GTF_RELOP_NAN_UN | GTF_UNSIGNED;
+            return gtFoldExpr(tmp);
+        }
+
+        case NI_SRCS_UNSAFE_IsAddressLessThan:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ldarg.1
+            // clt.un
+            // ret
+
+            GenTree* op2 = impPopStack().val;
+            GenTree* op1 = impPopStack().val;
+
+            GenTree* tmp = gtNewOperNode(GT_LT, TYP_INT, op1, op2);
+            tmp->gtFlags |= GTF_RELOP_NAN_UN | GTF_UNSIGNED;
+            return gtFoldExpr(tmp);
+        }
+
+        case NI_SRCS_UNSAFE_IsNullRef:
+        {
+            assert(sig->sigInst.methInstCount == 1);
+
+            // ldarg.0
+            // ldc.i4.0
+            // conv.u
+            // ceq
+            // ret
+            CLANG_FORMAT_COMMENT_ANCHOR;
+
+#ifdef TARGET_64BIT
+            bool uns = true;
+#else
+            bool uns                    = false;
+#endif
+
+            GenTree* op1 = impPopStack().val;
+
+            GenTree* cns = gtNewIconNode(0, TYP_INT);
+
+#ifdef TARGET_64BIT
+            cns = gtNewCastNode(TYP_I_IMPL, cns, uns, TYP_U_IMPL);
+#endif
```

Comment:
```suggestion
            GenTree* op1  = impPopStack().val;
            GenTree* zero = gtNewIconNode(0, genActualType(op1));
```
(With corresponding changes below)

No need to create cast nodes when we can create the constant node directly.

Reply:
Fixed.